src/utils/miscellaneous.ts

```diff
@@ -178,3 +178,7 @@ export async function retry<T extends () => any>(fn: T, options?: RetryOptions):
 export function flattenMapOfLists<K, T>(map: Map<K, T[]>): T[] {
   return Array.from(map.values()).reduce<T[]>((acc, list) => acc.concat(list), [])
 }
+
+export function flattenMapOfSets<K, T>(map: Map<K, Set<T>>): T[] {
+  return Array.from(map.values()).reduce<T[]>((acc, set) => acc.concat(Array.from(set)), [])
```

Comment:
brain hurtee

Reply:
I think it's fine to do some line breaks :) 

```js
const sets = Array.from(map.values())
sets.reduce<T[]>((acc, set) => {
  return acc.concat([ set ])
}, [])
```

Reply:
Or maybe even:

const sets = Array.from(map.values())
sets.reduce<T[]>((acc, set) => {
  return [ ...acc, set ]
}, [])

Reply:
No longer needed, I'll remove this entirely.